web/src/services/auth/authProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import { Auth0Provider } from "@auth0/auth0-react";
+import React, { createContext, ReactNode } from "react";
+
+import type { AuthHook } from "./authHook";
+import { useAuth0Auth } from "./authOAuth";
+import { useCognitoAuth } from "./cognitoAuth";
+
+export const AuthContext = createContext<AuthHook | null>(null);
+
+const Auth0Wrapper = ({ children }: { children: ReactNode }) => {
+  const auth = useAuth0Auth();
+  return <AuthContext.Provider value={auth}>{children}</AuthContext.Provider>;
+};
+
+const CognitoWrapper = ({ children }: { children: ReactNode }) => {
+  const auth = useCognitoAuth();
+  return <AuthContext.Provider value={auth}>{children}</AuthContext.Provider>;
+};
+
+export const AuthProvider: React.FC<{ children?: ReactNode }> = ({ children }) => {
+  const authProvider = window.REEARTH_CONFIG?.authProvider;
+
+  if (authProvider === "auth0") {
+    const domain = window.REEARTH_CONFIG?.auth0Domain;
+    const clientId = window.REEARTH_CONFIG?.auth0ClientId;
+    const audience = window.REEARTH_CONFIG?.auth0Audience;
+
+    return domain && clientId ? (
+      <Auth0Provider
+        domain={domain}
+        clientId={clientId}
+        authorizationParams={{
+          audience: audience,
+          scope: "openid profile email offline_access",
+          redirect_uri: window.location.origin,
+        }}
+        useRefreshTokens
+        useRefreshTokensFallback
+        cacheLocation="localstorage">
+        <Auth0Wrapper>{children}</Auth0Wrapper>
+      </Auth0Provider>
+    ) : null;
+  }
+
+  if (authProvider === "cognito") {
+    // No specific provider needed for Cognito/AWS Amplify
+    return <CognitoWrapper>{children}</CognitoWrapper>;
+  }
+
+  return <>{children}</>; // or some default fallback
```

Comment:
Should `auth0` be used in default?

Reply:
Yes, https://github.com/reearth/reearth/blob/1b433311e75b446356e17222033eb67689bd709f/web/src/services/config/index.ts#L75

Reply:
So should we align the code like the following?
```
if(authProvider === "cognito") {
...
}
return "Use Auth0 in here"
```

Reply:
Hmm, Yeah I guess. I was mostly aiming to have it so that we might want to support something like 
Microsoft Azure Active Directory in the future.

Reply:
Hmm I see. Let's keep it.